ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/CommandClient.scala

```diff
@@ -180,16 +181,18 @@ final class CommandClient(
           throw new IllegalArgumentException(
             s"Failing fast on submission request of command ${commands.commandId} with invalid application ID ${commands.applicationId} (client expected $applicationId)"
           )
-        val updateDedupTime = commands.deduplicationTime.orElse(
-          Some(
-            Duration
-              .of(
-                config.defaultDeduplicationTime.getSeconds,
-                config.defaultDeduplicationTime.getNano,
-              )
-          )
-        )
-        r.copy(commands = Some(commands.copy(deduplicationTime = updateDedupTime)))
+        val updatedDeduplicationPeriod = commands.deduplicationPeriod match {
+          case DeduplicationPeriod.Empty =>
+            DeduplicationPeriod.DeduplicationTime(
+              Duration
+                .of(
+                  config.defaultDeduplicationTime.getSeconds,
+                  config.defaultDeduplicationTime.getNano,
+                )
```

Comment:
We've really gotta extract out some conversion functions somewhere.

Reply:
I do aggree, so far I've seen it done either like this inline, using implicit extension methods, using a `Conversion` object to do the conversion, I would love to decide on one way of doing things so that we know where to find the conversions when we need them.